tests/test_cli_compile.py

```diff
@@ -2582,6 +2582,37 @@ def test_input_formats(fake_dists, runner, make_module, fname, content):
     assert "extra ==" not in out.stderr
 
 
+@pytest.mark.network
+def test_error_in_pyproject_toml(fake_dists, runner, make_module, capfd):
+    """
+    Test that an error in pyproject.toml is reported.
+    """
+    fname = "pyproject.toml"
+    content = """
+            [build-system]
+            requires = ["flit_core >=2,<4"]
+            build-backend = "flit_core.buildapi"
+
+            [tool.flit.metadata]
+            module = "sample_lib"
+            author = "Vincent Driessen"
+            author-email = "me@nvie.com"
+            requires = ["small-fake-a==0.1"]
+    """
+    content += """
+            maintainer = ["Vincent Driessen <nvie>"]  # error is here
+    """
+    meta_path = make_module(fname=fname, content=content)
```

Comment:
```suggestion
    invalid_content = dedent(
            """\
            [project]
            invalid = "metadata"
            """
    )
    meta_path = make_module(fname=fname, content=invalid_content)
```

Let's simplify `pyptoject.toml` to not depend on flit.